src/java/org/apache/cassandra/locator/ReconnectableSnitchHelper.java

```diff
@@ -61,10 +64,22 @@ private void reconnect(InetAddressAndPort publicAddress, VersionedValue localAdd
         }
     }
 
+    /**
+     * {@link org.apache.cassandra.auth.IInternodeAuthenticator#authenticate(InetAddress, int)} method is deprecated to
+     * support certificate based authentication.
+     *
+     * Non-certificate based authenticator implementations can do IP/PORT check in authenticate method.
+     *
+     * Certificate based authenticators can let authenticate method succeed if connection is REDIRECT as there are no
+     * certificates at this point to verify. Authentication check happens again in
+     * {@link org.apache.cassandra.net.OutboundConnectionInitiator.ServerAuthenticationHandler} after connection
+     * establishment in certificate based auth.
+     */
```

Comment:
This comment block is more about the implementation inside `reconnect`, not information callers of the snitch method would need. Better to document RECONNECT behavior at the enum, or the default method for `IInternodeAuthenticator.authenticate`.